File: DSA_Level-1/Recursion and Backtracking/Recursion with Arraylists/Get Subsequence/GetSubsequence_Solution.js

function getSubsequence(str){

    if(str.length==0){
        let res=new Array();
        res.push("");
        return res;
    }

    let ch=str[0];
    let subStr=str.substring(1);

    let thatRes=getSubsequence(subStr);
    let thisRes=new Array();

    for(let s of thatRes){
        thisRes.push(""+s);
    }

    for(let s of thatRes){
        thisRes.push(ch+s);
    }

    return thisRes;
}

console.log(getSubsequence("abc"));

File: DSA_Level-1/Recursion and Backtracking/Recursion with Arraylists/Get Subsequence/GetSubsequence_Solution.java

import java.io.*;
import java.util.*;

public class Main {

  public static void main(String[] args) throws Exception {
    Scanner scn = new Scanner(System.in);
    String str = scn.next();
    ArrayList< String> res = gss(str);
    System.out.println(res);
  }

  public static ArrayList<String> gss(String str) {

    if (str.length() == 0) {
      ArrayList<String> empty = new ArrayList<String>();
      empty.add("");
      return empty;
    }

    char ch = str.charAt(0);
    String subStr = str.substring(1);

    ArrayList<String> thatRes = gss(subStr);
    ArrayList<String> thisRes = new ArrayList<>();

    for (String s : thatRes) {
      thisRes.add("" + s);
    }

    for (String s : thatRes) {
      thisRes.add(ch + s);
    }

    return thisRes;
  }

}